mayastor/src/bdev/nexus/nexus_metadata.rs

```diff
@@ -5,821 +5,745 @@
-    }
 
-    /// Retrieve latest config object from "MetaData" partition.
-    pub async fn get_latest_config_object(
-        &self,
-        metadata: &NexusMetaData,
-    ) -> Result<Option<NexusConfig>, MetaDataError> {
-        if metadata.header.used_entries > 0 {
-            return Ok(Some(
-                self.probe_config_object(
-                    metadata,
-                    metadata.header.used_entries - 1,
-                )
-                .await?,
-            ));
-        }
-        Ok(None)
-    }
+        let mut list: Vec<MetaDataObject> = Vec::with_capacity(used as usize);
 
-    /// Remove selected config object from "MetaData" partition.
-    /// The "selected" parameter identifies the appropriate entry in the index
-    /// array.
-    pub async fn delete_config_object(
-        &mut self,
-        metadata: &mut NexusMetaData,
-        selected: u32,
-    ) -> Result<(), MetaDataError> {
-        if metadata.index.len() != metadata.header.used_entries as usize {
-            return Err(MetaDataError::IndexInconsistent {});
-        }
+        let start = if used > index.current_entry + 1 {
+            index.total_entries + index.current_entry + 1 - used
+        } else {
+            index.current_entry + 1 - used
+        };
+
+        if start > index.current_entry {
+            for offset in start .. index.total_entries {
+                handle
+                    .read_at((index.start_lba + offset) * block_size, &mut buf)
+                    .await
+                    .context(ReadError {
+                        name: String::from("object"),
+                    })?;
+                list.push(NexusMetaData::read_object(&buf)?);
+            }
 
-        if selected < metadata.header.used_entries {
-            metadata.index.remove(selected as usize);
-            metadata.header.used_entries -= 1;
+            for offset in 0 ..= index.current_entry {
+                handle
+                    .read_at((index.start_lba + offset) * block_size, &mut buf)
+                    .await
+                    .context(ReadError {
+                        name: String::from("object"),
+                    })?;
+                list.push(NexusMetaData::read_object(&buf)?);
+            }
+        } else {
+            for offset in start ..= index.current_entry {
+                handle
+                    .read_at((index.start_lba + offset) * block_size, &mut buf)
+                    .await
+                    .context(ReadError {
+                        name: String::from("object"),
+                    })?;
+                list.push(NexusMetaData::read_object(&buf)?);
+            }
         }
 
-        self.sync_metadata(metadata).await
+        Ok(list)
     }
 
-    /// Append a new config object to "MetaData" partition.
-    pub async fn append_config_object(
-        &mut self,
-        metadata: &mut NexusMetaData,
-        config: &NexusConfig,
+    /// Purge oldest entries from index.
```

Comment:
This doc comment might be improved with details about what the return value is, such as:

```rust
/// Purge oldest entries from index.
///
/// Returns the number of entries removed.
```

Reply:
Agreed. Will expand the description for the public functions.